Projects/Server/Common/Filters/IFilter.cs

```diff
@@ -0,0 +1,6 @@
+namespace Server.Common.Filters;
+
+public interface IFilter<in TIn, out TOut>
```

Comment:
Can we use an `Action<>` directly?

Reply:
If we use a function directly we will incur in heap allocations. With the struct filters we can implements filters even with closures (that cannot be cached by default) without allocations.
I made the "PredicateFilter" for fallback to normal Predicate or Func if needed by developers.
I know that create an ad-hoc struct for every case where we need a filter will produces many boilerplate code than a normal function creates a boilerplate code.
We can even switch to Func but the overhead of concatenation could be noticeable
.
As you suggested maybe it's better to switch to a dedicated library for LINQ source-generated when needed, that is surely more optimized and complete than my implementation.